Complete order management homework

YBSOG opened the pull request.

YBSOG committed:
Complete order management homework

Osch-1 commented on Order_Manager/order_manager.csproj:
```diff
@@ -0,0 +1,18 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
```
проекты называем в паскаль кейсе OrderManager
https://www.theserverside.com/definition/Pascal-case

Osch-1 commented on Order_Manager/Program.cs:
```diff
@@ -0,0 +1,76 @@
+﻿public class OrderManager
```
класс в котором есть метод Main лучше назвать Program
c# сейчас позволяет не объявлять этот класс и метод, но если делаете - лучше использовать стандартный подход

Osch-1 commented on Order_Manager/Program.cs:
```diff
@@ -0,0 +1,76 @@
+{
+    public static void Main()
+    {
+        OrderCreation();
```
методы называем глаголами - так как это начало вашей программы - можно назвать Run

Osch-1 commented on Order_Manager/Program.cs:
```diff
@@ -0,0 +1,76 @@
+        OrderCreation();
+    }
+
+    public static string CheckStringInput( string userStringInput )
```
1. в данном случае лучше поменять название - сейчас оно не соответствует тому что делает метод
можно назвать GetValidString
2. лучше разделить чтение и валидацию 
``` csharp
public static string GetValidInput()
{
  do
  {
    string readStr = Console.ReadLine();
  } while (!IsInputValid(readStr))
}


```

Osch-1 commented on Order_Manager/Program.cs:
```diff
@@ -0,0 +1,76 @@
+        int count;
+        string address;
+        string answer;
+        DateTime todays_date = DateTime.Now;
```
переменные называем в camelCase
https://www.freecodecamp.org/news/snake-case-vs-camel-case-vs-pascal-case-vs-kebab-case-whats-the-difference/

Osch-1 replied:
в данном случае `todays_date` -> `todaysDate`

Osch-1 commented on Order_Manager/Program.cs:
```diff
@@ -0,0 +1,76 @@
+        }
+    }
+
+    public static void OrderCreation()
```
получился очень большой метод - выделите чтение каждого из параметров в отдельные методы

Osch-1 commented on Order_Manager/Program.cs:
```diff
@@ -0,0 +1,76 @@
+        int count;
+        string address;
+        string answer;
+        DateTime todays_date = DateTime.Now;
```
лучше работать с датами в формате utc

Osch-1 commented on Order_Manager/Program.cs:
```diff
@@ -0,0 +1,76 @@
+        Console.WriteLine( "Введите необходимое количество единиц товара (цифрой от 1 до 9):" );
+        while ( true )
+        {
+            var countInput = Console.ReadLine();
```
var -> тип данных

Osch-1 commented on Order_Manager/Program.cs:
```diff
@@ -0,0 +1,76 @@
+        product = CheckStringInput( Console.ReadLine() );
+
+        Console.WriteLine( "Введите необходимое количество единиц товара (цифрой от 1 до 9):" );
+        while ( true )
```
лучше не использовать while true, старайтесь применять управляемы флаг - например типо bool
так проще понять по каким причинам данный цикл завершается и сложнее написать бесконечный цикл

Osch-1 reviewed (changes requested):
хороший код, есть комментарии по улучшению

YBSOG committed:
Исправления

Разделил все запросы по методам, исправил формат времени, название проекта, косяки с переменными. Изменил логику валидации вводимых строк.
